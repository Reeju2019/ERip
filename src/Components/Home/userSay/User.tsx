import React from 'react'
import { Container } from 'react-bootstrap'
import './user.css'
import data from '../../../Data/const.data.json'
import userData from '../../../Data/User.data.json'
import Carousel from 'react-multi-carousel'
import 'react-multi-carousel/lib/styles.css'

const User = () => {
  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 4,
      slidesToSlide: 1, 
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 2,
      slidesToSlide: 1,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 2,
      slidesToSlide: 1, 
    },
  }

  return (
    <div className='user_section'>
      <Container>
        {data.usersay.map((c, index) => {
          return (
            <div key={index}>
              <h4 className='user_heading d-flex justify-content-center'>{c.heading}</h4>
              <Carousel
                swipeable={true}
                draggable={true}
                showDots={true}
                responsive={responsive}
                ssr={true}
                infinite={true}
                autoPlay={false}
                autoPlaySpeed={1000}
                keyBoardControl={true}
                customTransition='all .5'
                transitionDuration={500}
                containerClass='carousel-container'
                removeArrowOnDeviceType={['tablet', 'mobile']}
                dotListClass='custom-dot-list-style'
                itemClass='carousel-item-padding-40-px'
                partialVisible={true}
                rewindWithAnimation={true}
                sliderClass='react-multi-carousel-track'
              >
                {userData.user.map((e, index) => {
                  return (
                    <div key={index} className='card user_card'>
                      <div className='img_card user_img'>
                        <img src={c.quotation} className='card-img-top' alt='...' />
                        <p className='user_card_text'>{e.text}</p>
                        <div className='d-flex'>
                          <img className='user_dp' src={e.user} alt='' />
                          <div className='user_id'>
                            <p className='card-text user_text'>{e.username}</p>
                            <p className='user_company'>
                              <i>-{e.company}</i>
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  )
                })}
              </Carousel>
            </div>
          )
        })}
      </Container>
    </div>
  )
}

export default User
