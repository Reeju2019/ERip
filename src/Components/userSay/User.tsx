import React from "react";
import { Container } from "react-bootstrap";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import { BsArrowRightShort, BsArrowLeftShort } from "react-icons/bs";
import { userData } from "./userData";
import "./user.css"

const User = () => {
  const settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 4,
    slidesToScroll: 1,
    prevArrow: <BsArrowLeftShort />,
    nextArrow: <BsArrowRightShort />,
    initialSlide: 0,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };
  return (
    <div id="device_section">
      <Container>
        <div>
          <h4 style={{ color: "#333946", fontWeight: "bold" }}>
            {" "}
            What Our Users Say{" "}
          </h4>
          <Slider {...settings}>
            {userData.map((e: any) => {
              return (
                <div className="card user_card">
                  <div className="img_card" style={{padding: "20px" }}>
                    <img
                      style={{ height: "15px", width: "20px"}}
                      src={e.image}
                      className="card-img-top"
                      alt="..."
                    />

                    <p
                      style={{
                        fontSize: "13px",
                        textAlign: "left",
                        paddingBottom: "40px",
                        paddingTop:"20px"
                      }}
                    >
                      {e.text}
                    </p>

                    <div className="d-flex">
                      <img
                        src={e.user}
                        alt=""
                        style={{ height: "50px", width: "50px", borderRadius:"50%", marginRight:"15px"}}
                      />
                      <div style={{textAlign:"left"}}>
                        <p className="card-text user_text" style={{fontSize:"13px", color:"#0e62cb", fontWeight:"600"}}>{e.username}</p>
                        <p style={{fontSize:"13px", color:"#646464"}}>
                          <i>-{e.company}</i>
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              );
            })}
          </Slider>
        </div>
      </Container>
    </div>
  );
};

export default User;
