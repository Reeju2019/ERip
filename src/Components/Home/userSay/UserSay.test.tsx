import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import User from './User'

test('user say', () => {
  render(<User />)
})

describe('pnf rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<User />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<User />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
