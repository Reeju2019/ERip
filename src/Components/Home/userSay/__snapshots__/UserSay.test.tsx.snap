// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pnf rendering specification SnapShot testing 1`] = `
<div
  className="user_section"
>
  <div
    className="container"
  >
    <div>
      <h4
        className="user_heading d-flex justify-content-center"
      >
        What Our Users Say
      </h4>
      <div
        className="react-multi-carousel-list carousel-container "
        dir="ltr"
      >
        <ul
          className="react-multi-carousel-track react-multi-carousel-track"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "overflow": "unset",
              "transform": "translate3d(0px,0,0)",
              "transition": "none",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
